packages/@tailwindcss-upgrade/src/migrate-js-config.ts

```diff
@@ -0,0 +1,121 @@
+import fs from 'node:fs/promises'
+import { dirname } from 'path'
+import type { Config } from 'tailwindcss'
+import { fileURLToPath } from 'url'
+import { loadModule } from '../../@tailwindcss-node/src/compile'
+import {
+  keyPathToCssProperty,
+  themeableValues,
+} from '../../tailwindcss/src/compat/apply-config-to-theme'
+import { info } from './utils/renderer'
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = dirname(__filename)
+
+export async function migrateJsConfig(fullConfigPath: string): Promise<void> {
+  let [unresolvedConfig, source] = await Promise.all([
+    loadModule(fullConfigPath, __dirname, () => {}).then((result) => result.module) as Config,
+    fs.readFile(fullConfigPath, 'utf-8'),
+  ])
+
+  if (!isSimpleConfig(unresolvedConfig, source)) {
+    info(
+      'The configuration file is not a simple object. Please refer to the migration guide for how to migrate it fully to Tailwind CSS v4. For now, we will load the configuration file as-is.',
+    )
+    return
+  }
+
+  let cssConfigs: string[] = []
+
+  if ('content' in unresolvedConfig) {
+    cssConfigs.push(migrateContent(unresolvedConfig as any))
+  }
+
+  if ('theme' in unresolvedConfig) {
+    cssConfigs.push(await migrateTheme(unresolvedConfig as any))
+  }
+
+  console.log(cssConfigs.join('\n'))
+}
+
+async function migrateTheme(unresolvedConfig: Config & { theme: any }): Promise<string> {
+  let { extend: extendTheme, ...overwriteTheme } = unresolvedConfig.theme
+
+  let resetNamespaces = new Set()
+
+  let css = `@theme reference inline {\n`
+  for (let [key, value] of themeableValues(overwriteTheme)) {
+    if (typeof value !== 'string' && typeof value !== 'number') {
+      continue
+    }
+
+    if (!resetNamespaces.has(key[0])) {
+      resetNamespaces.add(key[0])
+      css += `  --${keyPathToCssProperty([key[0]])}-*: initial;\n`
+    }
+
+    css += `  --${keyPathToCssProperty(key)}: ${value};\n`
+  }
+
+  for (let [key, value] of themeableValues(extendTheme)) {
+    if (typeof value !== 'string' && typeof value !== 'number') {
+      continue
+    }
+
+    css += `  --${keyPathToCssProperty(key)}: ${value};\n`
+  }
+
+  return css + '}\n'
+}
+
+function migrateContent(unresolvedConfig: Config & { content: any }): string {
+  let css = ''
+  for (let content of unresolvedConfig.content) {
+    if (typeof content !== 'string') {
+      throw new Error('Unsupported content value: ' + content)
+    }
+
+    css += `@source  "${content}";\n`
+  }
+  return css
+}
```

Comment:
One issue here is that we don't know what CSS file this need to be relative too. I guess this means we can't generate the CSS here and it will instead need to be a config object of some sorts that we send to the CSS migration.

Also: Is there any way to find out if the glob here would already be covered by the auto detection glob? We do have access to the resolved config globs here but I don't know how we can do a correct check here. I think this means the glob need to be expanded first? 🤔